pkg/core/mempool/mem_pool.go

```diff
@@ -69,6 +69,8 @@ type Pool struct {
 
 	resendThreshold uint32
 	resendFunc      func(*transaction.Transaction, interface{})
+	// removeStaleCallback is a callback method which is called after item is removed from the mempool.
+	removeStaleCallback func(*transaction.Transaction, interface{})
```

Comment:
I don't really like this one. Maybe we can avoid this if we're to save module data in transaction-associated data. Or maybe there is something else we can do. Of course addition/removal can be treated as events, but then there needs to be some more generic subscription logic (and we don't have any other users for it).

Reply:
The case is that we have to watch over all fallback transactions removed from mempool (not only for those which are removed due to main transaction appliance), so I'm not sure how it can be implemented without adding such callbacks/notifications. I replaced this callback by notifications from https://github.com/nspcc-dev/neo-go/pull/1658#discussion_r558285250.

> (and we don't have any other users for it)
I also think that Oracle module can make use of it for better requests maintainance.

Reply:
So fixed by mempool notifications.